cli/integration_tests/composable_config/composing-override-values-deps.t

```diff
@@ -42,3 +42,14 @@ Setup
    Tasks:    3 successful, 3 total
   Cached:    0 cached, 3 total
     Time:\s*[\.0-9]+m?s  (re)
+
+  $ ${TURBO} run override-values-task-with-deps-2 --filter=override-values --dry=json | jq '.tasks | map(select(.taskId == "override-values#override-values-task-with-deps-2")) | .[0].resolvedTaskDefinition'
+  {
+    "outputs": [],
+    "cache": true,
+    "dependsOn": [],
```

Comment:
As of now, this `dependsOn` is `["^override-values-underlying-topo-task"]`, which is the config from the root config. Not sure why the override from the workspace config doesn't overwrite it.

Reply:
Works fine with other array configs like `inputs`. Must be due to the custom unmarshaling of dependsOn